.github/workflows/test.yml

```diff
@@ -0,0 +1,135 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Tests
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [2.7, 3.9]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 pytest
+        pip install 'requests>=1.0' 'python-dateutil>=2.1' 'six>=1.2.0'
+        pip install pytest-cov python-coveralls
```

Comment:
Declare test dependencies in `setup.py` `extras_require` clause, then do `pip install '.[dev]'` here.

This is also useful for local development

Reply:
the goal of this PR is to move logic from travis more or less intact
your suggestions are nice but I think should be done in separate PR.